lib/invoker/power/power.rb

```diff
@@ -1,3 +1,31 @@
 require "invoker/power/http_response"
 require "invoker/power/dns"
 require "invoker/power/balancer"
+
+require "invoker/power/tld"
+
+module Invoker
+  module Power
+    class << self
+      def set_tld(tld_value)
+        @tld_value = tld_value
+      end
+
+      def reset_tld
+        @tld_value = nil
+      end
+
+      def tld
+        tld_value = @tld_value
+        if !tld_value && Invoker.config
+          tld_value = Invoker.config.tld
+        end
+        Tld.new(tld_value)
+      end
+
+      def tld_value
+        tld.value
+      end
+    end
+  end
```

Comment:
`tld` is also used in the `UrlRewriter` class https://github.com/nisanth074/invoker/tree/784916d1e643128ccaa9332472e8133daa67db9f/lib/invoker/power. So, I decided to not store `tld` in the `Setup` class and instead store it in the `Power` class.

I don't mind having `tld` as a simple string (instead of an object) if it makes the code more readable. Will change it.